### DISCUSSION TITLE
How can I pass CORS session cookies in cordova ios without using cordova-plugin-webview-proxy ?

### DISCUSSION THREAD

**Discussion by ghevge** [apache/cordova#405] [Q&A]
Hi,
I'm trying to sends some CORS requests from my ios cordova app to my backend server, keeping the session at the same time, but so far, i was only able to achieve this , using cordova-plugin-webview-proxy.

My problem with using cordova-plugin-webview-proxy is that, i have to define in the config.xml the following entires:
```
<preference name="scheme" value="app" />
<preference name="hostname" value="testapp"/>
```

the issue with this, is that when schema and host are defined, the  `navigator.mediaDevices` will always return undefined, so the camera won't be accessible. And I need the camera too in my app.

So what I need is an approach in which camera and CORS session calls are available.

The way I'm calling the Backend is : 
       
```
        settings["mode"] = 'cors';
	settings["credentials"] = 'include';
	let result = {};
        try {
            const response = await fetch(getFullURL(url), settings);
        ...

function getFullURL(url) {
    if (!('BASE_URL' in window) || isValueEmpty(BASE_URL)) {
        return url;
    } else {
        let temp = url.replace('./', BASE_URL);
        if (isIOSApp()) {
        	let result = window.WebviewProxy.convertProxyUrl(temp);
        	return result;
        } else {
			return temp;
		}

```    }
}

**Comment by ghevge** [comment]
Found a solution, which consists in  setting hostname = localhost.

```
<preference name="scheme" value="app" />
<preference name="hostname" value="localhost"/>
```

**Reply by lowright** [reply]
and now u can call api whithout cors ?

**Reply by breautek** [reply]
All browser requests will be subjected to CORs. There isn't a way to disable it.

So you either need to implement the CORs protocol on the server-side (e.g. the `OPTIONS` request and the `Access-Control` headers)

Or do not use browser fetch/xmlhttprequest APIs and instead use a cordova plugin that does HTTP requests using the native APIs.
